hermes/validator/question_generator.py

```
@@ -124,12 +140,74 @@ async def try_with_fallback():
+                return "", None, "No available remote challenges", None
+
+            first_candidate = None
+            second_candidate = None
+            third_candidate = None
+            for ch in chs:
+                if ch.id > self.remote_played_pivot and ch.id not in self.remote_played:
+                    first_candidate = ch
+                    break
+
+                if self.remote_played.get(ch.id, 1) < ch.max_count:
+                    third_candidate = third_candidate if third_candidate is None else third_candidate
```

Ruff (0.15.7)

[warning] Useless `if`-`else` condition

(RUF034)